Allows host config to override wamp_server and wamp_http

jlashner opened the pull request:
A quality of life improvement to allow us to override the wamp_server and wamp_http on a per-host basis (resolving #119). This is needed as the wamp_server and wamp_http addresses differ depending on whether the host is a docker container or not. For instance, crossbar may be accessible through `localhost:8001` if you're on the host, while being accessible through `sisock-crossbar:8001` if within a docker. This means users need to know where an OCS agent or client is being run to know the correct command line arguments to use, and must manually specify the server and http arguments pretty much every time they start an agent or a client.

I had to change `site_config.get_config` a bit, as it was previously not getting the host config dict for control objects, but I've tested my changes and made sure that it still works for both agents and clients. Let me know if there are any issues, thanks!

mhasself commented on ocs/site_config.py:
```diff
@@ -437,27 +437,29 @@ def get_config(args, agent_class=None):
+    if 'wamp_server' in host_config.data and args.site_hub is None:
+        site_config.hub.data['wamp_server'] = host_config.data['wamp_server']
+    if 'wamp_http' in host_config.data and args.site_http is None:
+        site_config.hub.data['wamp_http'] = host_config.data['wamp_http']
```
The 10 lines from here up will fault if host_config is None, which is completely allowed (for control clients for example)... please restore the guard.

jlashner commented:
Hey Matthew I think I fixed the bug you mentioned. I moved it so all site_config.host updates happen within the `if host_try in site_config.hosts` block, which I think addresses your concern about host_config not being defined for the updates. 

I also moved it to be before the command line args update statements since we want those to take priority. Let me know if there's anything else you think I should change!
